app/Http/Controllers/Api/ReportbackController.php

```diff
@@ -23,15 +24,28 @@ public function __construct(ReportbackService $reportbackService)
     }
 
     /**
-     * Store a newly created resource in storage.
-     * @todo upsert reportbacks by default.
+     * Store a newly created resource in storage or
+     * update a reportback if it already exists.
+     *
      * @param  \Illuminate\Http\Request  $request
      * @return \Illuminate\Http\Response
      */
-    public function create(Request $request)
+    public function store(ReportbackRequest $request)
     {
-        $reportback = $this->reportbackService->create($request->all());
+        // @TODO - instead should probably just have a method that gets northstar_id by default from a drupal_id if that is the only thing provided and then use that to find the reportback.
+        $userId = $request['northstar_id'] ? $request['northstar_id'] : $request['drupal_id'];
+        $type = $request['northstar_id'] ? 'northstar_id' : 'drupal_id';
 
-        return $this->item($reportback);
+        $reportback = $this->reportbackService->exists($request['campaign_id'], $request['campaign_run_id'], $userId, $type);
+
+        if (! $reportback) {
+            $reportback = $this->reportbackService->create($request->all());
+
+            return $this->item($reportback);
+        } else {
+            $updatedReportback = $this->reportbackService->update($reportback, $request->all());
+
+            return $this->item($updatedReportback);
```

Comment:
Another nitpicky suggestion, but these two `return`s could be pulled out from the conditional so there's always a "happy path" to the return at the end (and then it's just returning the same `$reportback` either way).